test/monitor.js

```diff
@@ -284,14 +284,61 @@ describe('good', function () {
 
                 monitor = new Monitor(new Hapi.Server(), options);
                 monitor.start(Hoek.ignore);
-            }).to.throw('reporter must specify events to filter on');
+            }).to.throw('reporter must specify events to filter on (0)');
 
             expect(function () {
 
                 options.reporters[0].events = { log: '*' };
                 monitor = new Monitor(new Hapi.Server(), options);
                 monitor.start(Hoek.ignore);
-            }).to.throw('Every reporter object must have an init method');
+            }).to.throw('Every reporter object must have an init method (0)');
+
+            done();
+        });
+
+        it('uses reporter name or index in reporter validation errors', function (done) {
+
+            var monitor;
+            var options = {};
+
+            var ignore = function () {};
+            ignore.attributes = {
+                name: 'test-reporter'
+            };
+
+            var one = {
+                reporter: ignore
+            };
+
+            options.reporters = [one];
+
+            expect(function () {
+
+                monitor = new Monitor(new Hapi.Server(), options);
+                monitor.start(Hoek.ignore);
+            }).to.throw('reporter must specify events to filter on (test-reporter)');
+
+            ignore.attributes = {
+                pkg: {
+                    name: 'test-reporter-two'
+                }
+            };
+
+            expect(function () {
+
+                monitor = new Monitor(new Hapi.Server(), options);
+                monitor.start(Hoek.ignore);
+            }).to.throw('reporter must specify events to filter on (test-reporter-two)');
```

Comment:
Can we change the message to it reads "test-reporter-two must specify events to filter on" and if there isn't a name, "reporter [1] must specify events to filter on" and have the number be 0-based like the JS array?